ibis/backends/materialize/__init__.py

```diff
@@ -0,0 +1,2832 @@
+                f"format must be 'pandas', 'arrow', or 'polars', got {format!r}"
+            )
+
+        # Import appropriate libraries based on format
+        if format == "pandas":
+            import pandas as pd
+        elif format == "arrow":
+            import pyarrow as pa
+        elif format == "polars":
+            try:
+                import polars as pl
+            except ImportError:
+                raise ImportError(
+                    "polars is required for format='polars'. "
+                    "Install it with: pip install polars"
+                ) from None
+
+        # Build SUBSCRIBE SQL
+        if isinstance(obj, str):
+            sql_parts = [f"SUBSCRIBE {obj}"]
+        else:
+            # It's an Ibis expression
+            self._run_pre_execute_hooks(obj)
+            query = self.compiler.to_sqlglot(obj)
+            sql_parts = [f"SUBSCRIBE ({query.sql(self.dialect)})"]
+
+        # Add envelope if specified
+        if envelope is not None:
+            envelope_upper = envelope.upper()
+            if envelope_upper not in ("UPSERT", "DEBEZIUM"):
+                raise ValueError("envelope must be 'UPSERT' or 'DEBEZIUM'")
+            sql_parts.append(f"ENVELOPE {envelope_upper}")
+
+        # Build WITH options
+        options = []
+        if not snapshot:
+            options.append("SNAPSHOT = FALSE")
+        if progress:
+            options.append("PROGRESS = TRUE")
+        if options:
+            sql_parts.append(f"WITH ({', '.join(options)})")
+
+        # Add AS OF timestamp
+        if as_of is not None:
+            sql_parts.append(f"AS OF {as_of}")
+
+        # Add UP TO timestamp
+        if up_to is not None:
+            sql_parts.append(f"UP TO {up_to}")
+
+        sql = " ".join(sql_parts)
+
+        # SUBSCRIBE needs a dedicated connection since it blocks
+        # Create a new connection with the same parameters
+        import psycopg
+
+        con_params = self.con.info.get_parameters()
+        sub_con = psycopg.connect(
+            host=con_params.get("host", "localhost"),
+            port=int(con_params.get("port", 6875)),
+            dbname=con_params.get("dbname", "materialize"),
+            user=con_params.get("user"),
+            password=con_params.get("password")
+            if hasattr(self.con, "_password")
+            else None,
+            autocommit=False,  # Need transaction for DECLARE CURSOR
+        )
+
+        cursor = sub_con.cursor()
+
+        try:
+            # Begin transaction and declare cursor for subscription
+            cursor.execute("BEGIN")
+            cursor_name = "ibis_sub_cursor"
+            cursor.execute(f"DECLARE {cursor_name} CURSOR FOR {sql}")
+
+            # Get column names after declaring cursor
+            columns = None
+
+            # Fetch results in batches
+            while True:
+                # Fetch a batch of rows
+                cursor.execute(f"FETCH {batch_size} {cursor_name}")
```

Comment:
Is there no way to use the pattern `while batch := cur.fetchmany(batch_size)` here?